src/check/parse/tokenize/src/main.zig

```diff
@@ -0,0 +1,1399 @@
+                        tok = T.Int;
+                        break;
+                    },
+                    'b', 'B' => {
+                        maybeMessageForUppercaseBase(self, c);
+                        self.pos += 1;
+                        self.chompIntegerBase2();
+                        self.chompNumberSuffix();
+                        tok = T.Int;
+                        break;
+                    },
+                    '0'...'9' => {
+                        self.pushMessageHere(.LeadingZero);
+                        _ = self.chompNumberBase10();
+                        self.chompNumberSuffix();
+                        tok = T.Int;
+                        break;
+                    },
+                    '_' => {
+                        self.pos += 1;
+                        continue;
+                    },
+                    '.' => {
+                        self.pos += 1;
+                        _ = self.chompIntegerBase10();
+                        tok = T.Float;
+                        _ = self.chompExponent();
+                        break;
+                    },
+                    else => {
+                        tok = T.Int;
+                        break;
+                    },
+                }
+            }
+        } else {
+            _ = self.chompNumberBase10();
+            self.chompNumberSuffix();
+            tok = T.Int;
+        }
+        return tok;
+    }
+
+    pub fn chompExponent(self: *Cursor) bool {
+        if (self.peek() orelse 0 == 'e' or self.peek() orelse 0 == 'E') {
+            self.pos += 1;
+            // Optional sign
+            if (self.peek() orelse 0 == '+' or self.peek() orelse 0 == '-') {
+                self.pos += 1;
+            }
+            self.chompIntegerBase10();
+            return true;
+        }
+        return false;
+    }
+
+    pub fn chompNumberSuffix(self: *Cursor) void {
+        if (self.peek() == null or !std.ascii.isAlphabetic(self.peek() orelse 0)) {
+            return;
+        }
+        const start = self.pos;
+        var pos: usize = self.pos + 1;
+        while (pos < self.buf.len) : (pos += 1) {
+            const c = self.buf[pos];
+            if (std.ascii.isAlphabetic(c) or std.ascii.isDigit(c)) {
+                // continue advancing
+            } else {
+                break;
+            }
+        }
+        const suffix = self.buf[start..pos];
+        if (!(std.mem.eql(u8, suffix, "dec") or
+            std.mem.eql(u8, suffix, "i128") or
+            std.mem.eql(u8, suffix, "i16") or
+            std.mem.eql(u8, suffix, "i32") or
+            std.mem.eql(u8, suffix, "i64") or
+            std.mem.eql(u8, suffix, "i8") or
+            std.mem.eql(u8, suffix, "nat") or
+            std.mem.eql(u8, suffix, "u128") or
+            std.mem.eql(u8, suffix, "u16") or
+            std.mem.eql(u8, suffix, "u32") or
+            std.mem.eql(u8, suffix, "u64") or
+            std.mem.eql(u8, suffix, "u8")))
```

Comment:
There must be a nicer and more efficient way to do this? Maybe use a static string map like we do for matching keywords?